lib/yabeda/test_adapter.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require "singleton"
+
+require_relative "./base_adapter"
+
+module Yabeda
+  # Fake monitoring system adapter that collects latest metric values for later inspection
+  class TestAdapter < BaseAdapter
+    include Singleton
```

Comment:
If we plan to add Mintiest support in the future, it's probably worth considering some kind of per-thread storage (for multithread tests).